libs/cardano-ledger-api/src/Cardano/Ledger/Api/State/Query.hs

```diff
@@ -207,9 +211,19 @@ queryCommitteeMembersState coldCredsFilter hotCredsFilter statusFilter nes = do
                 Nothing -> MemberNotAuthorized
                 Just Nothing -> MemberResigned
                 Just (Just hk) -> MemberAuthorized hk
-        -- TODO: implement after pulsing and snapshots are done
-        let nextEpochChange = NoChangeExpected
-        pure $ CommitteeMemberState hkStatus status mbExpiry nextEpochChange
+        pure $ CommitteeMemberState hkStatus status mbExpiry (nextEpochChange coldCred)
+
+      nextEpochChange :: Credential 'ColdCommitteeRole (EraCrypto era) -> NextEpochChange
+      nextEpochChange ck
+        | inCurrent && not inNext = ToBeRemoved
```

Comment:
We need to look at all the relevant keys to report if they will be removed or not
```suggestion
        | Map.member ck relevantColdKeys && not inNext = ToBeRemoved
```

Reply:
Actually I think in the new implementation, `not inNext` is sufficient  to determine that it has to be removed  (since it means: it's not in current and also not in next)